### PR TITLE
Match chrome (header and sidebar nav) to new docs

### PR DESCRIPTION
**Pull request by mperrotti**
Closes https://github.com/github/primer/issues/4166

We should wait until the new docs site is launched before merging this. **Update 02 Dec 2024:** soft launch is happening sometime this week - likely Friday 06 Dec 2024.

## Summary

Updates the design of the docs site navigation to match the new Primer docs site (https://github.com/github/primer-docs) to give a more "seamless" feel when navigating between the different sub-sites.

We chose not to make these changes in doctocat to avoid accidentally breaking things downstream.

## List of notable changes:

- Header navigation has a new layout
- Search input moved to the sidebar and functions like the search input in the new Primer docs.

## What should reviewers focus on?

Confirm that this matches the design of the new Primer docs site.

## Steps to test:

1. Run the docs site
2. Look at the navigation at various viewports

## Supporting resources (related issues, external links, etc):

-
-

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

_Note: these have become outdated as we've iterated, but they're pretty close_

### Child pages

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2024-10-16 at 11 32 20 AM

 </td>
<td valign="top">

Screenshot 2024-10-16 at 11 19 11 AM

</td>
</tr>
<tr>
<td valign="top">

Screenshot 2024-10-16 at 11 32 34 AM

 </td>
<td valign="top">

Screenshot 2024-10-16 at 11 22 20 AM

</td>
</tr>
</table>

### Landing page

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2024-10-16 at 11 32 20 AM

 </td>
<td valign="top">

Screenshot 2024-10-16 at 11 19 11 AM

</td>
</tr>
<tr>
<td valign="top">

Screenshot 2024-10-16 at 11 32 34 AM

 </td>
<td valign="top">

Screenshot 2024-10-16 at 11 22 20 AM

</td>
</tr>
</table>

### CODE REVIEW

**Review by emilybrick — CHANGES_REQUESTED** [review]
Hey there! This looks awesome! Leaving some feedback from the deployment link. 

Screenshot 2024-10-21 at 12 05 34 PM

I believe this should be `Primer / Brand UI`, since that's the page the user is currently on, and Brand UI should not be listed to the right with the other links. 

Is it possible to update the side nav in this PR to match new docs as well? Main things missing is Mona Sans. If the section headers aren't clickable, I think they're fine left in gray/muted color.

**Review by emilybrick — APPROVED** [review]
This looks awesome! Thank you so much 🙌 
chrome-capture-2024-10-22 (1)

**Comment by emilybrick** [comment]
This looks great! I'm sorry, I mentioned in [Slack](https://github.slack.com/archives/D02B49AUA49/p1729710228723819) but I should have put it here - 


> can we match the max-width of the brand page to the proiduct UI page, and add margin: 0 auto? Right now, on wide screens, they are different widths. 

> The max-width of the product UI page is actually being calculated based on the cards below (each card component has a max width built in), but I think it ends up around...1340px max-width. I tested this in the inspector and that appears to be the right size. 


Screenshot 2024-10-23 at 10 28 53 PM
Screenshot 2024-10-23 at 10 28 56 PM

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/hero.js
Comment (rezrah) [inline apps/docs/src/@primer/gatsby-theme-doctocat/components/hero.js:15]: Can we make this a H1 to meet heading order guidelines please?

```suggestion
        <Heading as="h1" size="3">
```

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/hero.js
Comment (rezrah) [inline apps/docs/src/@primer/gatsby-theme-doctocat/components/hero.js:26]: To match the one used on Brand Toolkit.

```suggestion
          alt="3d illustration, showing a side profile of Mona the GitHub Mascot gazing towards the sky"
```

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/header.module.css
Comment (rezrah) [inline apps/docs/src/@primer/gatsby-theme-doctocat/components/header.module.css:27]: Size is really small. Switching to a Primer Brand token should fix this ☝️ 

```suggestion
  font-size: var(--brand-text-size-400);
```

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/header.js
Comment (rezrah) [inline apps/docs/src/@primer/gatsby-theme-doctocat/components/header.js:12]: Can we make this link to Primer Brand docs? Our users rarely need to go to the homepage or product docs so it feels intuitive to send them to the root. Ideally it would work like our marketing pages, where clicking on the title text takes you back to Primer Brand UI homepage and clicking the Logo icon will take you back to primer.style. What do you think?

https://github.com/user-attachments/assets/8960d252-667f-4bab-af66-4c11a1659fb5

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/header.module.css
Comment (rezrah) [inline apps/docs/src/@primer/gatsby-theme-doctocat/components/header.module.css:9]:
```suggestion
  border-bottom: var(--brand-borderWidth-thin) solid var(--brand-color-border-muted);
```

File: apps/docs/src/components/navigation/navigation.module.css
Comment (rezrah) [inline apps/docs/src/components/navigation/navigation.module.css:40]: FYI this color token isn't loaded on the page so looks like this 👇 

Screenshot 2024-12-03 at 09 59 09

Comment (rezrah) [reply apps/docs/src/components/navigation/navigation.module.css:40]: Suggestion to fix color and size to match new docs:

```suggestion
  color: var(--brand-color-text-muted);
  font-size: var(--brand-text-size-400);
```

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/header.module.css
Comment (rezrah) [inline apps/docs/src/@primer/gatsby-theme-doctocat/components/header.module.css:31]: Can we add some padding before the separator too, as it's too close the text:

```suggestion
  padding-inline-end: var(--base-size-12);
}
```

File: apps/docs/src/components/side-panel/side-panel.module.css
Comment (rezrah) [inline apps/docs/src/components/side-panel/side-panel.module.css:7]: Close button is currently aligned to the start, can we move to end position instead?

```suggestion
  border-bottom: var(--brand-borderWidth-thin) solid var(--brand-color-border-muted);
  display: flex;
  justify-content: end;
```


<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">


Screenshot 2024-12-03 at 10 11 31

 </td>
<td valign="top">


Screenshot 2024-12-03 at 10 11 25

</td>
</tr>
</table>

File: apps/docs/src/components/side-panel/side-panel.tsx
Comment (rezrah) [inline apps/docs/src/components/side-panel/side-panel.tsx:53]: Is this something you'll be looking at straight after @mperrotti or shall we take this on in Brand? This workaround feels quite broken with the double scroll bars and centering. 


https://github.com/user-attachments/assets/6c612b02-8c04-446d-b241-9e1947b12bca

Comment (mperrotti) [reply apps/docs/src/components/side-panel/side-panel.tsx:53]: Double scrollbars should definitely be fixed. I can't think of a way around centering except:
- Upgrade to a newer version of `@primer/react`
- Attempt to use the `sx` prop to fake a side sheet

I'll try using the `sx` prop to manually style this like a side sheet.

File: apps/docs/src/components/side-panel/side-panel.module.css
Comment (mperrotti) [reply apps/docs/src/components/side-panel/side-panel.module.css:7]: Let me check what we do in PRC. I agree that the end position is much better.

Comment (mperrotti) [reply apps/docs/src/components/side-panel/side-panel.module.css:7]: End position is the correct placement. I'm also not sure why the IconButton isn't using the `"invisible"` variant. I may update that as well.

File: apps/docs/src/components/navigation/navigation.module.css
Comment (mperrotti) [reply apps/docs/src/components/navigation/navigation.module.css:40]: Very strange - I thought I already fixed the CSS vars in another commit. Let me make sure I didn't stash it somewhere by accident...

Comment (mperrotti) [reply apps/docs/src/components/navigation/navigation.module.css:40]: Can't find my changes, but I can take yours. Thank you!!

File: apps/docs/src/components/side-panel/side-panel.tsx
Comment (mperrotti) [reply apps/docs/src/components/side-panel/side-panel.tsx:53]: I ended up achieving this with the `sx` prop

File: apps/docs/src/components/navigation/navigation.tsx
Comment (rezrah) [inline apps/docs/src/components/navigation/navigation.tsx:17]: Should we use consistent ordering given that this is a navigation menu shared across all docs sites?

E.g. in product docs, the product link comes first, but in brand docs the brand link is first. I'd expect them to stay the same so perhaps they don't need to be sorted at all?

Screenshot 2024-12-05 at 14 29 31


Screenshot 2024-12-05 at 14 26 11

File: apps/docs/src/components/side-panel/side-panel.tsx
Comment (rezrah) [reply apps/docs/src/components/side-panel/side-panel.tsx:53]: Looks and works great on mobile now, thanks @mperrotti

File: apps/docs/src/components/side-panel/side-panel.tsx
Comment (rezrah) [inline apps/docs/src/components/side-panel/side-panel.tsx:61]: Brand tokens are prefixed

```suggestion
            borderRadius: 'var(--brand-borderRadius-large, 0.75rem)',
```

File: apps/docs/src/components/side-panel/side-panel.tsx
Comment (rezrah) [inline apps/docs/src/components/side-panel/side-panel.tsx:69]: We fortunately have equivalent tokens for these...

```suggestion
              'Overlay--motion-slideInLeft var(--brand-animation-duration-faster) var(--brand-animation-easing-glide) 0s 1 normal none running',
```

**Review by rezrah — APPROVED** [review]

File: apps/docs/src/components/navigation/navigation.tsx
Comment (mperrotti) [reply apps/docs/src/components/navigation/navigation.tsx:17]: Forgot to change this when we changed the design of the top nav. Pushing up a fix.

Comment (emilybrick) [reply apps/docs/src/components/navigation/navigation.tsx:17]: Agreed - @mperrotti and I actually spoke about this last night. Will fix before it gets merged.

**Comment by emilybrick** [comment]
Note for follow up (I don't think we need to add it to this PR) - let's have consistent hover states on the top nav items. 

Product UI nav item hover state:
Screenshot 2024-12-05 at 11 04 00 AM

Brand UI nav items don't have a hover state currently ☝️

**Review by emilybrick — APPROVED** [review]
Approving assuming we can fix the suggestions mentioned. Thank you!!

Left a small comment here: https://github.com/primer/brand/pull/790#issuecomment-2520777680

**Comment by mperrotti** [comment]
@rezrah - any idea what could be causing this NPM error in CI?

```
Run npx json -I -f cra-test/tsconfig.json -e 'this.exclude=["node_modules", "**/*.cy.ts"]'
npm warn exec The following package was not found and will be installed: json@11.0.0
json: error: ENOENT: no such file or directory, open 'cra-test/tsconfig.json'
```

File: apps/docs/src/@primer/gatsby-theme-doctocat/components/header.js
Comment (raytalks) [reply apps/docs/src/@primer/gatsby-theme-doctocat/components/header.js:12]: @mperrotti 👋🏽 After a conversation with @tallys, we’d like to send the users to the root, as suggested by @rezrah, and insulate Primer Brand from Primer to reduce dependencies.

**Comment by rezrah** [comment]
> @rezrah - any idea what could be causing this NPM error in CI?

They're unrelated to your changes @mperrotti - feel free to merge your PR when ready.

Several integration tests coincidentally started failing at the same time for various reasons. I'll take a look at fixing those later this week.

**Pull request merged**
